### ISSUE TITLE
API problems

### ISSUE THREAD

**Issue by ff1992ff** [SkyTubeTeam/SkyTube#664]
Dear Skytube team, 
I had to update from the old relative stable build without API problems to the recent one, because in channel view no videos were displayed in most cases since Friday... Right after installing I got an API quote exceeded error message. However, I created a fresh individual YouTube API key and it worked for 3 hours. Then it said the daily quota is exceeded... I mean, I have 400 channels, but an API only used by me should not be capped that fast. Then I tried an old API that I created a few years ago for KODI, but although the key was correct, it was not able to verify - neither the old one nor the new one. I also forced stopped execution of the app in order to restart, but no difference. I also tried Newpipe backend, but that didn't work either. Can you help me? 

Another thing that is annoying me with the new version is, that the GUI language always is my native language (German) and not English, which I prefer... 

I don't want to switch to Newpipe as Skytube is in my opion far superior in terms of the design and functionality than Newpipe, Vanced and the official build...

PS: By the way maybe a suggestion for future builds: the current build of the YouTube KODI addon has, if I remember correctly 10 API keys included after installation with the option to add personal ones... Maybe an option to distribute the user load of the integrated API key...

Anyway, thank you for your help and for the great work!

**Comment by okan35** [comment]
I am also not happy with api quota exceeded issue as I am 100% sure, I don't watch that many videos in such short time or it just gives the message for the newly created key. 

I haven't checked and I just shooting the idea I have, this quota exceeded issue might be related to fetching videos for the feed or because of some similar process so we could somehow use all the quota.

**Comment by gzsombor** [comment]
I've found the quota exceeded problem very strange indeed. Unfortunately I couldn't reproduce it locally - I mean, I saw the error messages, about the limit, but in the logs, I couldn't find any unusual API usage, which was higher than expected. Have you checked the detailed statistics about what API method is used? It is visible here: https://console.developers.google.com/apis/api/youtube.googleapis.com/metrics?project=YOUR_PROJECT_ID (just change the YOUR_PROJECT_ID to your project id :wink: )
 Currently, SkyTube use 6 API keys, so the idea is good, unfortunately, it is not enough :+1:  
My intention is to move as many calls to the Newpipe backend, as possible. That backend is getting better, and I'm really curious, what didn't work for you?
 Unfortunately, when the API key validation fails, due to quota limit error - the app refuse to use that key. Maybe, we could add a dialog about, if the user want to force-use the invalid key?
 I can't remember if there were any changes in the locale handling, do you know since when it's behave this way? Maybe, the appcompat/androidx changes somehow caused this...

**Comment by ff1992ff** [comment]
Here is a link to my API stats... Sorry it is in German, but I assume the GUI elements should be at the same place anywhere in the world:

https://imgur.com/a/3BKHl2F

**Comment by ff1992ff** [comment]
Sorry for double posting: imgur compressed the image that much, that you can hardly read anything... Here is an upload to another image hoster:

https://ibb.co/3WQJ4pk

**Comment by 13r0ck** [comment]
The solution for me was to go to console.developers.google.com and generate an API key for the YouTube Data API v3. Then enter it into the app under Preferences -> Others -> YouTube API Key

**Comment by ff1992ff** [comment]
Hi 13r0ck... Thanks for your suggestion. I already did that, but after 2 - 3 hours it also always gives me the quota exceeded warning. After that I'm not able to verify any other API key, until it is reset again...

**Comment by gzsombor** [comment]
Sorry for the late reply, unfortunately understanding Youtube API behavior is far from me, I don't understand, why it gave you the error about the quota usage, as you only sent 656 requests. I'm wondering, maybe, because it happened in a very short timeframe?

**Comment by avently** [comment]
@gzsombor
I have this problem with quota too. That's strange because I don't have too much channels (30-40). For example, today I entered in the app 5-6 times, used a search once, and now when I tried to open a channel I got quota error. Can you look at this problem closer? Are you able to reproduce it on your own?

**Comment by ff1992ff** [comment]
Hi again! It seems gzsombor was right and there were too many calls in a short time frame. I suppose, when I imported my backup (between 400-500 channels), the API interpreted it as a DDOS attack or something. 

I updated now to the latest version of Skytube Extra and basically import every single channel by hand, in order to not trigger the API quota alert.

I have a suggestion for the developers; large channel backups like mine do probably mess with the API. Would it be possible to implement a new backup format, that basically divides the large dataset into smaller chunks (e.g. 30-40 channels per chunk) in order to distribute the API calls in a way, that they don't get the quota exceeded error.

For example:
Backup (300 channels) = 10 chunks (30 channels)
This is how I would imagine an Import process: 
restore chunk 1 ---> cooldown 60 seconds ---> restore chunk 2 ---> cooldown ---> ... ---> restore chunk 10

Would something like that be possible? Thank you for your help and suggestions!
